packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -39,13 +38,72 @@ import { CommandLineOptions, ShellCommandBuilder } from '@theia/process/lib/comm
 
 export const TERMINAL_WIDGET_FACTORY_ID = 'terminal';
 
+/**
+ * Terminal widget options.
+ */
+export const TerminalWidgetOptions = Symbol('TerminalWidgetOptions');
+export interface TerminalWidgetOptions {
+
+    /**
+     * Human readable terminal representation on the UI.
+     */
+    readonly title?: string;
+
+    /**
+     * Path to the executable shell. For example: `/bin/bash`, `bash`, `sh`.
+     */
+    readonly shellPath?: string;
+
+    /**
+     * Shell arguments to executable shell, for example: [`-l`] - without login.
+     */
+    readonly shellArgs?: string[];
+
+    /**
+     * Current working directory.
+     */
+    readonly cwd?: string;
+
+    /**
+     * Environment variables for terminal.
+     */
+    readonly env?: { [key: string]: string | null };
+
+    /**
+     * In case `destroyTermOnClose` is true - terminal process will be destroyed on close terminal widget, otherwise will be kept
+     * alive.
+     */
+    readonly destroyTermOnClose?: boolean;
+
+    /**
+     * Terminal server side can send to the client `terminal title` to display this value on the UI. If
+     * useServerTitle = true then display this title, otherwise display title defined by 'title' argument.
+     */
+    readonly useServerTitle?: boolean;
+
+    /**
+     * Terminal id. Should be unique for all DOM.
+     */
+    readonly id?: string;
+
+    /**
+     * Terminal attributes. Can be useful to apply some implementation specific information.
+     */
+    readonly attributes?: { [key: string]: string | null };
+
+    /**
+     * Terminal kind that indicates whether a terminal is created by a user or by some extension for a user
+     */
+    readonly kind?: 'user' | string;
+}
+
 export interface TerminalWidgetFactoryOptions extends Partial<TerminalWidgetOptions> {
     /* a unique string per terminal */
     created: string
 }
 
 @injectable()
-export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget {
+export class TerminalWidget extends BaseWidget implements StatefulWidget {
```

Comment:
we should move everything to `@theia/terminal/lib/browser/terminal-widget` to align with our guidelines

we can keep `terminal-widget-impl` for now, but it should only reexport

Reexported runtime variables should be deprecated, that downstream receive a warning that it was broken and can be removed in future minor releases.